includes/class-blocks.php

```diff
@@ -21,12 +21,15 @@ public static function init() {
 		// This is already being called on the init hook, so just add it.
 		self::register_blocks();
 
-		\add_action( 'wp_enqueue_scripts', array( self::class, 'add_data' ) );
-		\add_action( 'enqueue_block_editor_assets', array( self::class, 'add_data' ) );
+		\add_action( 'wp_enqueue_scripts', array( self::class, 'inject_activitypub_options' ) );
+		\add_action( 'admin_print_scripts', array( self::class, 'inject_activitypub_options' ) );
 		\add_action( 'load-post-new.php', array( self::class, 'handle_in_reply_to_get_param' ) );
 		// Add editor plugin.
 		\add_action( 'enqueue_block_editor_assets', array( self::class, 'enqueue_editor_assets' ) );
 		\add_action( 'init', array( self::class, 'register_postmeta' ), 11 );
+		if ( ! \wp_is_block_theme() ) {
```

Comment:
@obenland FYI https://github.com/Automattic/wordpress-activitypub/blob/8770b5b3b8c3a696ca8e0f919e9b1cf2acf82a93/includes/functions.php#L571

Reply:
what about setting the complete init of the class behind `if ( ! \wp_is_block_theme() ) {`???

Reply:
> Because if I deactivate the setting, I will not be able to place it somewhere else

Why not? Should it not behave like any other block?

Reply:
Let me ask it differently: How can I add the block somewhere else in the hybrid theme, without having it added to the post?

Reply:
Because the setting has a different meaning in the different modes, I can not add the block if the setting is disabled. But if I enable the setting, the code adds the block to the `the_content` hook, because it "thinks" it is a classic theme!

Reply:
I would keep the block-template completely out of that. Enabling the setting should only be to add/remove it from the `the_content` hook... like a `legacy` mode...

The per post setting on the other hand can work for both.